Code/Tools/FBuild/FBuildCore/Graph/DependencyListNode.cpp

```diff
@@ -0,0 +1,243 @@
+            {
+                Collect( dep.GetNode() );
+            }
+        }
+    }
+
+    bool IsMatch( const char * filename ) const
+    {
+        for ( const AString & pattern : Patterns )
+        {
+            if ( PathUtils::IsWildcardMatch( pattern.Get(), filename ) )
+            {
+                return true;
+            }
+        }
+        return Patterns.IsEmpty();
+    }
+
+    bool DumpTxtFile( const char* filename ) const
+    {
+        ASSERT( filename );
+
+        // Collect all file dependencies
+
+        Array< const AString * > dependencyList;
+
+        for ( const bucket_t & bucket : Visiteds )
+        {
+            for ( const Node * node : bucket )
+            {
+                if ( node->IsAFile() && IsMatch( node->GetName().Get() ) )
+                {
+                    dependencyList.Append( &node->GetName() );
+                }
+            }
+        }
+
+        // Sort collected filenames
+
+        dependencyList.Sort([]( const AString * a, const AString * b ) {
+            return ( a->Compare( *b ) < 0 );
+        });
+
+        // Prepare output buffer
+
+        AStackString<> buffer;
+
+        for ( const AString * depname : dependencyList )
+        {
+            buffer += *depname;
+            buffer += "\r\n";
+        }
+
+        // Dump to txt file
+
+        FileStream dmp;
+        if ( dmp.Open( filename, FileStream::WRITE_ONLY ) == false )
+        {
+            return false;
+        }
+        else
+        {
+            dmp.WriteBuffer( buffer.Get(), buffer.GetLength() * sizeof(*buffer.Get()) );
+            dmp.Close();
+
+            return true;
+        }
+    }
+
+    static bool MakeListTxt( const Node * node, const char * filename, const Array< AString > & patterns )
+    {
+        DependencyListBuilder builder( patterns );
+
+        // Don't want to collect root node itself
+
+        builder.Collect( node->GetStaticDependencies() );
+        builder.Collect( node->GetDynamicDependencies() );
+
+        return builder.DumpTxtFile( filename );
+    }
+};
+
+// CONSTRUCTOR
+//------------------------------------------------------------------------------
+DependencyListNode::DependencyListNode()
+: FileNode( AString::GetEmpty(), Node::FLAG_NONE )
+{
+    m_Type = Node::DEPENDENCY_LIST_NODE;
+}
+
+// Initialize
+//------------------------------------------------------------------------------
+/*virtual*/ bool DependencyListNode::Initialize( NodeGraph & nodeGraph, const BFFToken * funcStartIter, const Function * function )
+{
+    // .PreBuildDependencies
+    if ( !InitializePreBuildDependencies( nodeGraph, funcStartIter, function, m_PreBuildDependencyNames ) )
+    {
+        return false; // InitializePreBuildDependencies will have emitted an error
+    }
+
+    // Get node for Source of dependency list
+    Node* srcNode = nodeGraph.FindNode( m_Source );
+    if ( srcNode != nullptr )
+    {
+        m_StaticDependencies.Append(Dependency( srcNode ));
+    }
```

Comment:
Failure to retrieve the node should be an error - since it is a misconfiguration.

This is the preferred mechanism for retrieving a dependency, which handles error output for you:
```C++
    Dependencies compilerExeFile( 1, false );
    if ( !Function::GetFileNode( nodeGraph, iter, function, m_Source, ".Source", m_StaticDependencies ) )
    {
        return false; // GetFileNode will have emitted an error
    }
```